src/main/webapp/app/components/newCompanyForm/NewCompanyForm.tsx

```diff
@@ -0,0 +1,302 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import { observable, action } from 'mobx';
+import {
+  AvField,
+  AvForm,
+  AvGroup,
+  AvInput,
+  AvFeedback,
+} from 'availity-reactstrap-validation';
+import { Button, Col, Row } from 'react-bootstrap';
+import {
+  LicenseType,
+  LICENSE_TITLES,
+  LicenseStatus,
+  CompanyType,
+  COMPANY_TYPE_TITLES,
+  LICENSE_STATUS_TITLES,
+  LicenseModel,
+  LICENSE_MODEL_TITLE,
+} from 'app/config/constants';
+import { getSectionClassName } from 'app/pages/account/AccountUtils';
+import autobind from 'autobind-decorator';
+import client from 'app/shared/api/clientInstance';
+import {
+  CompanyDTO,
+  CompanyDomainDTO,
+  CompanyVM,
+} from 'app/shared/api/generated/API';
+import styles from './styles.module.scss';
+
+type INewCompanyFormProps = {
+  onValidSubmit: (newCompany: Partial<CompanyVM>) => void;
+};
+
+type IDomainListBoxProps = {
+  domainList: string[];
+  updateDomainList: (actionType: DomainListActions, event: any) => void;
+};
+
+enum DomainListActions {
+  ADD,
+  DELETE,
+}
+
+@observer
+class DomainListBox extends React.Component<IDomainListBoxProps> {
+  render() {
+    return (
+      <div className={styles.domain_box}>
+        <div className={styles.main}>
+          {this.props.domainList.map((domain, idx) => (
+            <div key={idx} className={styles.content_container}>
```

Comment:
I don't think using index as the key is recommended. https://reactjs.org/docs/lists-and-keys.html#keys
We can use domain here.